ZIPonline.js

```diff
@@ -0,0 +1,267 @@
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.indexOf('/heft')>-1 && url.indexOf('/zip')>-1) {
+		//Z.debug(ZU.xpathText(doc, '//meta[@property="article:section"]/@content'));
+		var articleSection = ZU.xpathText(doc, '//meta[@property="article:section"]/@content');
+		if (articleSection.indexOf('Rechtsprechung')>-1) {
+			return "case";
+		}
+		return "journalArticle";
+		
+	} else if (url.indexOf('/archivsuche/')>-1 || url.indexOf('/aktuelles-heft/')>-1 || url.indexOf('/heft-')>-1) {
+		if (getSearchResults(doc, true)) {
+			return "multiple";
+		}
+	}
+	//search in fulltexts will only return a list with direct links to pdfs
+	//and are therefore not handled here
+	//https://www.zip-online.de/volltexte/
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//ul/li[contains(@class, "aufsatz") or contains(@class, "rechtsprechung")]/a[not(contains(@href, "//www."))]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var type = detectWeb(doc, url);
+	
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	//translator.setDocument(doc);
+
+	translator.setHandler('itemDone', function (obj, item) {
+		var authors = ZU.xpath(doc, '//span[@class="beitrag-autor"]');
+		if (authors) {
+			item.creators = [];
+			var firstName, lastName;
+			for (var i=0; i<authors.length; i++) {
+				firstName = ZU.xpathText(authors[i], './/span[contains(@class, "beitrag-autorvorname")]');
+				lastName = ZU.xpathText(authors[i], './/span[contains(@class, "beitrag-autornachname")]');
+				if (firstName && lastName) {
+					item.creators.push({'firstName': firstName, 'lastName': lastName, 'creatorType': "author"});
+				} else if (lastName && lastName.trim() !== "") {
+					item.creators.push({'lastName': lastName, 'creatorType': "author"});
```

Comment:
The three likely scenarios are 1) institutional author 2) bad data entry (first and lastname in same field) 3) missing first name, right? Given that using fieldMode=1 is correct for 1) and preferable for 2) since data can be fixed user-side with 1 click, I think we should do that. I think that's also our general practice in such cases. I don't feel particularly strongly about it, though, so if you have a strong opinion to the contrary I'll accept this as is.